src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/SubscriptionLocationService.java

```diff
@@ -85,6 +94,29 @@ public String deleteSubscriptionByLocation(String locationId, String provenanceU
 
     }
 
+    private void deleteAllSubscriptionListTypeForLocation(List<Subscription> locationSubscriptions) {
+        List<String> uniqueUsers = locationSubscriptions.stream()
+            .map(Subscription::getUserId).distinct().toList();
+        List<UUID> subscriptionListTypesIds = new ArrayList<>();
+        int minNoOfSubscriptions = 1;
+
+        if (!uniqueUsers.isEmpty()) {
+
+            for (String userId : uniqueUsers) {
+                List<Subscription> userLocationSubscriptions = repository.findLocationSubscriptionsByUserId(userId);
+                if (userLocationSubscriptions.size() <= minNoOfSubscriptions) {
+                    Optional<SubscriptionListType> subscriptionListType =
+                        subscriptionListTypeRepository.findByUserId(userId);
+                    subscriptionListType.ifPresent(listType -> subscriptionListTypesIds.add(listType.getId()));
```

Comment:
I think rather than using the deleteByIdIn on line 115 and therefore needing to store the ID's in an array / the if statement, it'd be simpler to just call the subscriptionListTypeRepository.delete() method here instead

Reply:
fixed.